discord/sonica-discord.py

```diff
@@ -6,26 +6,113 @@
 from discord.ext import commands
 
 from sonica_pb2_grpc import SonicaStub
-from sonica_pb2 import Empty
+from sonica_pb2 import Empty, Search, Status
 
 bot = commands.Bot(command_prefix='')
-channel = grpc.insecure_channel('localhost:7700')
-print("Trying to connect to Sonica daemon...")
-# TODO: Implement some timeout wait if the daemon isn't running or something goes wrong
-daemon = SonicaStub(channel)
+
+def surround(surround, string):
+    return f'{surround}{string}{surround}'
+
+def songformat(song):
+    return surround('**', song.title) + ' - ' + surround('*', song.artist)
 
 @bot.command()
 async def play(ctx):
-    res = daemon.Play(Empty())
-    await ctx.message.channel.send(f'result: {str(res.success)}')
+    r = daemon.Play(Empty())
+    if not r.success:
+        await ctx.message.channel.send(r.reason)
+
+@bot.command()
+async def stop(ctx):
+    r = daemon.Stop(Empty())
+    if not r.success:
+        await ctx.message.channel.send(r.reason)
+
+@bot.command()
+async def skip(ctx):
+    r = daemon.Skip(Empty())
+    if not r.success:
+        await ctx.message.channel.send(r.reason)
+
+enumerators = {}
+
+@bot.command()
+async def search(ctx, *queries):
+    global enumerators
+    query = ' '.join(queries)
+    q = [query]
+    r = daemon.Search(Search.Query( query = q ))
+
+    # Note this is the same code as in the cli, so any bugs here will be shared there
+    mes = 'Here you go ^-^\n'
+    counter = 0
+    map = {}
+    for e in r.results:
+        mes += surround('**', e.name) + '\n'
+        if len(e.possibilities) == 0:
+            mes += 'Nothing\n'
+        else:
+            for id, song in e.possibilities.items():
+                counter += 1
+                mes += f'{counter}: {songformat(song)}\n'
+                map[counter] = (id, song)
+
```

Comment:
```suggestion
                map[counter] = (id, song)
        mes += '\n'
```
Add a newline between each engine result.